test/support/utils.h

```diff
@@ -1061,6 +1061,139 @@ get_pattern_for_test_sizes()
     return sizes;
 }
 
+template <typename T>
+struct IsMultipleOf
+{
+    T value;
+
+    bool operator()(T v) const
+    {
+        return v % value == 0;
+    }
+};
+
+template <typename T>
+struct IsEven
+{
+    bool
+    operator()(T v) const
+    {
+        if constexpr (std::is_floating_point_v<T>)
+        {
+            std::uint32_t i = (std::uint32_t)v;
+            return i % 2 == 0;
+        }
+        else
+        {
+            return v % 2 == 0;
+        }
+    }
+};
+
+template <typename T>
+struct IsOdd
+{
+    bool
+    operator()(T v) const
+    {
+        if constexpr (std::is_floating_point_v<T>)
+        {
+            std::uint32_t i = (std::uint32_t)v;
+            return i % 2 != 0;
+        }
+        else
+        {
+            return v % 2 != 0;
+        }
+    }
+};
+
+template <typename T>
+struct IsGreatThen
+{
+    T value;
+
+    bool
+    operator()(T v) const
+    {
+        return v > value;
+    }
+};
+
+template <typename T>
+struct IsLessThen
```

Comment:
nitpick: `IsLessThan` and `IsGreatThan` is probably the best name.

Reply:
Fixed, thanks